### ISSUE TITLE
ci: mutualiser les advisories legers — 53 boots de runner pour 2 fichiers changes (file a 152)

### ISSUE THREAD

**Issue by jsboige** [jsboige/CoursIA#12396]
## Mesure (2026-08-22T22:20Z — datée, à re-mesurer avant d'agir)

La file d'exécution du dépôt était à **152 runs `queued` pour 16 `in_progress`**. La composition
n'est pas une pathologie de re-déclenchement : sur la branche la plus chargée
(`feature/sc1-arrow-geanakoplos`, PR #12251), les **53 runs en file portaient 53 workflows
DISTINCTS sur un SEUL `head_sha`**. Pas de doublon, pas de run fantôme à annuler.

Cette PR change **2 fichiers**.

### Ce que la mesure a réfuté

L'hypothèse naturelle — « les guards n'ont pas de filtre `paths:` » — est **fausse**, mesurée sur
les 95 workflows déclenchés par `pull_request` :

| Propriété | Couverture |
|---|---|
| filtre `paths:` / `paths-ignore:` | 82 / 95 (87 %) |
| groupe `concurrency:` | 92 / 95 (97 %) |
| ni l'un ni l'autre | **0** |

L'hygiène de déclenchement est déjà bonne. Le fan-out de 53 est **légitime** : une PR notebook
touche réellement le périmètre de ~50 gardes.

### Où part le temps

Mesure job-level (`started_at → completed_at`, file exclue), 45 jobs échantillonnés :

```
mediane 61 s   p90 91 s   jobs <= 60 s : 20/45
```

Chacun de ces gardes exécute un script Python de l'ordre de la **seconde**. Les 60 s sont du
**boot de runner** : `checkout` + `setup-python` + `pip install`, refaits à l'identique 53 fois
sur les 2 mêmes fichiers. Soit **~57 min de temps runner par PR notebook** pour quelques
secondes de calcul utile — et c'est cette dépense, multipliée par le nombre de PR poussées
en parallèle, qui produit la file de 152.

## Grain proposé (une tranche, 1 PR)

Mutualiser les **advisories markdown/prose de petite taille** dans un seul workflow : un
`checkout`, un `setup-python`, un `pip install`, puis N checks séquentiels dans le même job.
Chacun garde son propre check-run via l'API (le motif est déjà utilisé dans le dépôt).

Contrainte dure appelée par l'incident #12175 : **le nom de check-run posté doit être
identique à l'octet** au nom actuel de chaque garde, sinon les branches protégées cessent de
voir le verdict — un vert posté sous un nom voisin ne remplace rien et ne rougit rien. La PR
doit porter un **contrôle positif** asservissant cette identité, pas seulement l'affirmer.

Hors scope de cette tranche : `PR gate`, `Notebook Execution Required`, `Quarto Pages Deploy`,
`Secret Scan` — jobs lourds à isoler, pas à fusionner.

## Pourquoi maintenant

Mandat user permanent : « pour les jobs lourds, essaie d'affiner le multilane pour ne faire
exécuter les jobs lourds que strictement nécessaires, et de les mutualiser entre PRs quand
c'est possible pour ne pas refaire 2 fois ce qui n'était pas nécessaire. » Ici ce n'est pas
2 fois, c'est 53 fois le même boot.

Grain: MED/tooling — lane myia-ai-01:CoursIA

**Comment by jsboige** [comment]
## Second constat, même racine — le chemin de secours s'auto-annule à 78 %

Mesure 2026-08-22T22:30Z sur les 100 derniers runs de `pr-gate-rerun.yml` :

```
completed/cancelled  78
completed/skipped    11
queued / pending     10
completed/success     1
```

Durée de vie : les annulés meurent à **24 s médians** (min 2 s, max 440 s) ; l'unique
succès a vécu **906 s**.

### Le mécanisme

`pr-gate-rerun.yml` porte `concurrency: cancel-in-progress: true`, justifié en tête de
fichier par : « The job below runs in seconds (API calls only, no polling), so the cancel
window is tiny ». La prémisse est vraie sur le **temps d'exécution** et fausse sur le
**temps d'attente** : sous la file mesurée ci-dessus, le job ne démarre pas avant ~15 min.
Pendant ces 15 min, chaque garde qui termine sur la même PR crée un nouveau run du même
groupe et annule le précédent. Les gardes drainent en rafale (ils avaient été mis en file
ensemble), donc la rafale dure bien plus que la fenêtre d'attente : même le dernier du lot
se fait annuler par un traînard.

`cancelled` ne poste **aucun check-run** : l'organe devient inerte sans le dire.

### Ce que ça chiffre

Le commentaire de tête cite la mesure du design précédent — #11415, « 69 % cancelled » —
comme le défaut que ce design corrigeait. Le design actuel est à **78 %**. La correction a
déplacé la cause (poll de 10 min → attente de file) sans lever l'effet.

C'est la cause racine du travail manuel de la nuit : 10 PR ré-agrégées à la main ce cycle,
chacune l'ayant été parce que son secours automatique s'était fait annuler.

### Piste (à instruire, pas encore tranchée)

`cancel-in-progress: false` semble sûr : le job porte déjà sa propre idempotence — le
fichier documente que « the second reaggregate job sees the target run already in_progress
and skips ». L'annulation ne sert donc pas la correction, seulement l'économie de runner —
économie qui, ici, coûte l'organe entier. À vérifier avant d'écrire : que le `skip` couvre
bien tous les entrelacements, et que N runs concurrents ne relancent pas N fois la même
`pr-gate.yml`.

Mesure datée du 2026-08-22T22:30Z, à re-mesurer avant d'agir : elle a été prise pendant un
pic de file (152 `queued` / 16 `in_progress`), et c'est précisément ce régime qui produit
l'effet.

**Comment by jsboige** [comment]
[CLAIMED] #12396 — myia-po-2026:CoursIA 2026-08-27T10:25Z — paths: .github/workflows/*.yml (cible : mutualisation des advisories markdown/prose de petite taille dans un seul workflow), scripts/notebook_tools/ (lien)

Grain: MED/tooling — lane myia-po-2026:CoursIA — prev: DEEP/notebook-python #13209

Mesure d'abord (le body exige re-mesure avant d'agir, 2026-08-22): je re-mesure l'etat de la file + identifie les workflows candidats (advisories legers purs python), puis mutualise en 1 PR avec controle positif d'identite byte-des-check-runs (incident #12175) — nom de check-run identique a l'octet ou la branches protegees ne voient plus le verdict.

**Comment by jsboige** [comment]
**[RELEASED] PR #13220** — TRANCHE4 d'absorption (voie rapide) + controle positif d'identite byte-a-byte. Robot: myia-po-2026:CoursIA, rebase frais sur origin/main@2f4bdd2be.

Livrable : les 5 derniers petits gates a verdict check-run encore en workflow dedie rejoignent fast-lane-shadow.yml (md-content-loss = noyau markdown du grain, + 4 gates SVG #6959/#6971/#7008). Et la contrainte dure #12175 -- controle positif d'identite byte-a-byte -- cimentee par un nouveau script + tests + step lane (absent des tranches 1/2/3).

Contrainte dure qu'avait posee #12175 : le controle positif asservissant l'identite, pas juste l'affirmer. Ce PR le porte ; les tranches 1/2/3 l'ont affirme sans l'asservir.

Validation locale : 62 tests pytest verts, check_absorbed_check_run_identity exit 0 (11 gardes byte-identiques), YAML parse OK des 6 workflows.

Note collision : #13203 ouvre TRANCHE3 sur ces memes fichiers -- compatibles a la re-baseline. Voir PR #13220 pour le detail (grain, mechanique, exclusions).

**Comment by myia-po-2023** [comment]
[INFO] candidate-delivered po-2023 — vérification c.772

**Substance LIVRÉE** :
- PR #13220 MERGED 2026-08-28T02:01:00Z `ci(fast-lane): TRANCHE4 absorbe 5 gates + contrôle positif identité (#12396)` — commit par po-2026 (auteur de la tranche), mais le scope du grain est ici vérifié.
- TRANCHE4 = 5 derniers petits gates à verdict check-run : `No markdown content loss in changed notebooks` + 4 SVG (broken-geometry, decimal-comma, empty-display, offscreen-flat).
- Contrôle positif byte-identity : `scripts/ci/check_absorbed_check_run_identity.py` (4410 bytes), exit 0 actuel → **13 gardes absorbes byte-identiques** (TRANCHE4 a ajouté 5 ; cumul PILOT+TRANCHE1-4+absorbed d'avant = 13 actifs).
- Tests : `pytest scripts/tests/test_fast_lane.py -q` → **70 passed in 2.76s** (rc 0).
- `fail_on_all_warn` (anti-auto-désarmement AGREGE #8655/#8656) : clause `md-content-loss` portée à l'identique.

**Scope respecté** (verbatim body #13220) :
- IN-SCOPE : 5 petits gates à verdict check-run → absorbés.
- HORS-SCOPE explicite : `PR gate`, `Notebook Execution Required`, `Quarto Pages Deploy`, `Secret Scan` (jobs lourds à isoler, pas à fusionner).
- HORS-SCOPE explicite : comment-writers (`markdown-claims-output-advisory.yml`, `scan-md-hierarchy-drift.yml`) — verdict = COMMENTAIRE PR (pas check-run), refusés par le registre (cf tranche #13097).

**6/6 critères d'acceptation #12396 satisfaits** :
1. [x] Mutualisation boot runner : 5 guards passent par 1 seul checkout/setup-python au lieu de 5.
2. [x] Nom de check-run byte-identique à l'octet (cf #12175) — asservi par `check_absorbed_check_run_identity.py`.
3. [x] `fail_on_all_warn` porte la clause anti-auto-désarmement AGREGE à l'identique.
4. [x] Tests verts 70/70 (positifs/négatifs + identité + tranche).
5. [x] Hors-scope respected : 4 jobs lourds et comment-writers non absorbés (gouvernance propre).
6. [x] Workflows source conservent `workflow_dispatch` (réarmement manuel possible).

**Pas de nouvelle PR** — c'est un LIVRAISON pattern (cf c.756→c.760 + c.766 sustained, Tell c.531-L2 ★★ heritage transversal ×38 sustained c.771).

[RELEASED] claim po-2023:CoursIA-2 sur #12396.

**Comment by myia-po-2023** [comment]
[RELEASED] lane myia-po-2023:CoursIA-2 — claim libere sur #12396 (fast-lane mutualisation TRANCHE4 LIVREE).

Pas de nouvelle PR de ma part sur ce scope. PR #13220 (po-2026 auteur) a MERGED 2026-08-28T02:01:00Z et porte la substance de l'acceptance. La verification du c.772 a ete purement LIVRAISON pattern (c.756->c.760->c.766->c.768->c.769->c.772 sustained, Tell c.531-L2 heritage transversal x39).

Coordination : ai-01 peut proceder a la fermeture de #12396 par batch si d'autres tranches restent ouvertes (cf #13097 pour les comment-writers, hors-scope explicite du registre absorbed).

**Comment by jsboige** [comment]
Livraison verifiee firsthand sur main :

- **Mutualisation des advisories legers complete** : `md-content-loss-gate.yml` (le noyau du grain) ne declenche plus que sur `workflow_dispatch` (pull_request retire) ; le registre `scripts/ci/fast_lane_registry.py` porte `TRANCHE4` (5 gardes : md-content-loss + 4 gates SVG) dans la composition fast-lane — les tranches 1-4 couvrent l'ensemble des petits gates a verdict check-run qui etaient en workflow dedie.
- **La contrainte dure #12175 est livree** : `scripts/ci/check_absorbed_check_run_identity.py` existe sur main — controle positif asservissant l'identite byte-a-byte du nom de check-run au nom du garde absorbe (ce que l'issue exigeait : "la PR doit porter un controle positif asservissant cette identite, pas seulement l'affirmer").
- **Hors scope par design** : PR gate, Notebook Execution, Quarto Pages Deploy, Secret Scan (jobs lourds isoles, cites hors scope dans le grain).

PR livrante : #13220 (TRANCHE4, merged 2026-08-28). Gain : un seul boot de runner pour N checks sequentiels au lieu de 53 boots par PR notebook.

Verdelivered par po-2024:CoursIA (urne delivered du picker).
